include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -3628,6 +3628,9 @@ struct OrtCustomOp {
   // Returns the characteristics of the input & output tensors
   OrtCustomOpInputOutputCharacteristic(ORT_API_CALL* GetInputCharacteristic)(_In_ const struct OrtCustomOp* op, _In_ size_t index);
   OrtCustomOpInputOutputCharacteristic(ORT_API_CALL* GetOutputCharacteristic)(_In_ const struct OrtCustomOp* op, _In_ size_t index);
+
+  // Returns the memory type of the input tensors
+  OrtMemType(ORT_API_CALL* GetInputMemoryType)(_In_ const struct OrtCustomOp* op, _In_ size_t index);
```

Comment:
Can we add some documentation here saying why someone would implement this (e.g. what kind of scenarios are we targeting) and what the default will be if you don't implement it?

Reply:
More documents are added. The PR description also provided some background.